### PR TITLE
Add service name header based on log name

### PR DESCRIPTION
**Pull request by sharbov**
Generated Hammock client will now accept service name parameter in its contractor
If set all requests will pass that name in a dedicated service-name header.
If it was not set the code will try to determine the service name based on the opened stratolog files.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ cliff==1.15.0
 coloredlogs==5.0
 mock==1.3.0
 cmd2==0.6.9
+psutil==2.2.1
```
Comment (ravidbro) [inline requirements.txt]: hammock is not using PBR, hence you need to add it also in the setup.py.
Other than that, nice work

File: hammock/templates/client/file.j2 [hunk 2]
```diff
@@ -217,6 +217,68 @@ class {{ class_name }}(object):
...
+    return headers
+
+
+def get_service_name():
+    """Return service name based on its log file name."""
+    try:
+        strato_logs = get_strato_log_names()
+        return find_common_prefix(strato_logs)
+    except:
+        return
```
Comment (igal-stratoscale) [inline hammock/templates/client/file.j2]: why not pass? maybe better some default value?

Comment (sharbov) [reply hammock/templates/client/file.j2]: I want to return `None` in this case. using `pass` will result in the same behaviour

File: hammock/templates/client/file.j2 [hunk 3]
```diff
@@ -217,6 +217,68 @@ class {{ class_name }}(object):
...
+def find_common_prefix(strings):
+    """Return the common prefix between the given strings."""
+    if len(strings) == 0:
+        return
+
+    if len(strings) == 1:
+        return strings[0]
+
+    common_characters = []
+    strings_count = len(strings)
```
Comment (igal-stratoscale) [inline hammock/templates/client/file.j2]: maybe better to move it to the first line and use strings_count in the ifs?

File: hammock/templates/client/file.j2 [hunk 4]
```diff
@@ -217,6 +217,68 @@ class {{ class_name }}(object):
...
+def get_service_name():
+    """Return service name based on its log file name."""
+    try:
+        strato_logs = get_strato_log_names()
+        return find_common_prefix(strato_logs)
+    except:
+        return
+
+
+def get_strato_log_names():
```
Comment (rom-stratoscale) [inline hammock/templates/client/file.j2]: btw, how would it behaves for melet for example (that it's no in a container)

File: hammock/templates/client/file.j2 [hunk 5]
```diff
@@ -217,6 +217,68 @@ class {{ class_name }}(object):
...
+            open_file_path = os.readlink(os.path.join(file_desc_dir, file_desc))
+            if open_file_path.endswith(".stratolog"):
+                strato_logs.append(os.path.basename(open_file_path)[:-10])
+        except OSError:
+            pass
+
+    return strato_logs
+
+
+def find_common_prefix(strings):
```
Comment (rom-stratoscale) [inline hammock/templates/client/file.j2]: os.path.commonprefix :)

File: hammock/templates/client/file.j2 (hunk 4 above)
Comment (sharbov) [reply hammock/templates/client/file.j2]: It **should** behave the same, the open files are associated by the current process id

Comment (rom-stratoscale) [reply hammock/templates/client/file.j2]: @sharbov how did u test it?

Comment (sharbov) [reply hammock/templates/client/file.j2]: by running it

**Pull request merged**